cmd/heartbeats/main.go

```diff
@@ -131,36 +134,72 @@ func main() {
 		log.Printf("Failed to read OIDC token, client will not send Authorization header: %v", err)
 	}
 
-	conf, err := config.JSONToTracingConfig(env.TracingConfig)
+	cfg := &observability.Config{}
+
+	err = json.Unmarshal([]byte(env.ObservabilityConfig), cfg)
```

Comment:
As above, you're not checking `err` here.

Given the similarities in the two sets of code, I wonder whether extracting them into a common method would be helpful, possibly with a signature like:

```
// SetUpObservability configures global otel meter and tracing providers, returning a
// cleanup function to be called on shutdown, and an error.
SetUpObservability(context.Context) (func(), error)
```

Reply:
Also, should this use `config.GetObservabilityConfig()`, rather than building its own `cfg` and calling `json.Unmarshal()` itself?

Reply:
(`adapter/v2/config` uses `envconfig`, which might be a blocker on this plan...)

Reply:
@evankanderson I created #8643 to capture this, I'd like to think a bit more about how to clean this up in the context of https://github.com/knative/pkg/issues/3216